model_analyzer/analyzer.py

```diff
@@ -129,20 +133,38 @@ def profile_model(self, run_config, perf_output_writer=None):
             config=self._create_perf_config(run_config))
 
         # Get metrics for model inference and write perf_output
-        model_metrics = self._profile(perf_analyzer=perf_analyzer,
-                                      dcgm_monitor=dcgm_monitor)
+        model_gpu_metrics, inference_metrics = self._profile(
+            perf_analyzer=perf_analyzer, dcgm_monitor=dcgm_monitor)
         if perf_output_writer:
             perf_output_writer.write(perf_analyzer.output() + '\n')
 
+        # Process GPU Metrics
+        gpu_metrics = defaultdict(list)
+        for _, metric in model_gpu_metrics.items():
+            for gpu_id, metric_value in metric.items():
+                gpu_metrics[gpu_id].append(metric_value)
+
+        for gpu_id, metrics in gpu_metrics.items():
+            # Model name here is triton-server, batch and concurrency
+            # are defaults
+            output_row = [
+                run_config['model-name'], run_config['batch-size'],
+                run_config['concurrency-range']
+            ]
+            output_row += metrics
+            output_row.append(gpu_id)
+            self._tables['model_gpu_metrics'].add_row(output_row)
+
+        # Process Inference Metrics
         output_row = [
             run_config['model-name'], run_config['batch-size'],
             run_config['concurrency-range']
         ]
+
         output_row += [
-            model_metrics[metric] for metric in self._monitoring_metrics
+            inference_metrics[metric] for metric in inference_metrics
```

Comment:
`output_row += inference_metrics.values()`  will make it clear that you're getting the values from a dict